src/main/java/com/exasol/adapter/dynamodb/documentfetcher/dynamodb/DynamodbScanDocumentFetcher.java

```diff
@@ -5,78 +5,109 @@
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
-import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
-import com.amazonaws.services.dynamodbv2.model.AttributeValue;
-import com.amazonaws.services.dynamodbv2.model.ScanRequest;
-import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import com.exasol.adapter.dynamodb.documentnode.DocumentValue;
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeVisitor;
+
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.ScanRequest;
+import software.amazon.awssdk.services.dynamodb.model.ScanResponse;
 
 /**
  * This class represents a DynamoDB {@code SCAN} operation.
  */
 class DynamodbScanDocumentFetcher extends AbstractDynamodbDocumentFetcher {
     private static final long serialVersionUID = 8061635540586030441L;
-    private final ScanRequest scanRequest;
+    private final GenericTableAccessParameters genericParameters;
+    private final int totalSegments;
+    private final int segment;
 
     /**
      * Create an instance of {@link DynamodbScanDocumentFetcher}.
-     *
-     * @param scanRequest DynamoDB scan request
+     * 
+     * @param genericParameters generic parameters (not scan specific)
+     * @param totalSegments     number of segments for the parallel scan
+     * @param segment           segment to scan
      */
-    DynamodbScanDocumentFetcher(final ScanRequest scanRequest) {
-        this.scanRequest = scanRequest;
+    private DynamodbScanDocumentFetcher(final GenericTableAccessParameters genericParameters, final int totalSegments,
+            final int segment) {
+        this.genericParameters = genericParameters;
+        this.totalSegments = totalSegments;
+        this.segment = segment;
+    }
+
+    static Builder builder() {
+        return new Builder();
     }
 
     ScanRequest getScanRequest() {
-        return this.scanRequest;
+        final ScanRequest.Builder templateScanRequest = ScanRequest.builder();
```

Comment:
I think builder should be named `builder`. Hard to understand the code below with this name

Reply:
The comment from this class also applicable to the second similar class